### PR TITLE
cam6_3_150: Adding convective gustiness to U10: Add UGUST output to CAM

### LINKED ISSUE #540: Convective gustiness impacts added to ocean surface flux #540
Related to[ CMEPS issue #267](https://github.com/ESCOMP/CMEPS/issues/267), we're exploring how to add the effects of convective gustiness into the calculation of surface fluxes over oceans. The code mods so far are all contained within CMEPS (`esmFldsExchange_cesm_mod.F90`, `med_phases_aofluxes_mod.F90`, and `shr_flux_mod.F90`) but the impacts of the addition also influence the atmosphere and might require some tuning. 

We've tested this with 32L (ten year F2010 runs, diagnostics here: [gust_32L_diag](https://webext.cgd.ucar.edu/AMWG/gust_mods/F2010climo.f09_f09_mg17.TestGustGlobal.001-F2010climo.f09_f09_mg17.TestGustGlobal_Control.001/)), and over a shorter period with 58L [(gust_58L_diag).](https://webext.cgd.ucar.edu/AMWG/gust_mods/F2010climo.f09_f09_mg17.TestGustGlobal_58L.001-F2010climo.f09_f09_mg17.TestGustGlobal_Control_58L.001/)  For reference, there are also comparisons to observations in the [parent directory](https://webext.cgd.ucar.edu/AMWG/gust_mods/).

### PR DESCRIPTION
**Pull request by megandevlan**
Closes #540 
This corresponds to issue https://github.com/ESCOMP/CAM/issues/540, and CMEPS https://github.com/ESCOMP/CMEPS/issues/267 (pull request for CMEPS side - https://github.com/ESCOMP/CMEPS/pull/422). 

Adding convective gusts to U10 occurs in the coupler, but the added magnitude (UGUST) needs to be output in CAM; these mods enable that. 

Run to confirm the impact of this change is at: https://github.com/NCAR/amwg_dev/issues/440

### CODE REVIEW

**Comment by cacraigucar** [comment]
This PR will need an update to Externals.cfg for the CMEPS external.  @jedwards4b says this is currently cmeps0.14.48, but recent discussions indicate that another tag may be made.

**Comment by megandevlan** [comment]
I believe the CMEPS external will now be cmeps0.14.49.

File: src/physics/cam/cam_diagnostics.F90 [hunk 1]
```diff
@@ -501,6 +501,7 @@ subroutine diag_init_moist(pbuf2d)
     call addfld ('TREFHTMX', horiz_only, 'X','K','Maximum reference height temperature over output period')
     call addfld ('QREFHT',   horiz_only, 'A', 'kg/kg','Reference height humidity')
     call addfld ('U10',      horiz_only, 'A', 'm/s','10m wind speed')
+    call addfld ('UGUST',      horiz_only, 'A', 'm/s','Gustiness term added to U10')
```
Comment (peverwhee) [inline src/physics/cam/cam_diagnostics.F90:504]: nitpick, but would be nice if this line (and 1789) can be adjusted to align with the above lines for readability

**Review by peverwhee — APPROVED** [review]
one nitpick, but looks good!

File: src/physics/cam/cam_diagnostics.F90 (hunk 1 above)
Comment (megandevlan) [reply src/physics/cam/cam_diagnostics.F90]: Thanks for catching that! Just updated my branch so that the lines are properly aligned now.

**Review by cacraigucar — CHANGES_REQUESTED** [review]
The CMEPS external needs to be added to this PR.

This appears to only be adding a diagnostic variable at the CAM level.  If that is the case, this is fine.  If it does somehow change answers for cam_dev runs (due to the CMEPS external update), then a way to turn this on/off via a namelist might need to be introduced.

**Comment by megandevlan** [comment]
The CAM mods are mostly to add the diagnostic variable, but it's true that for that to work I should have updated the externals -  I _think_ I've done that here. There's a namelist option to turn the gusts on/off, which is set in usr_nl_cpl (add_gust)

**Review by cacraigucar — APPROVED** [review]

**Pull request merged**